src/BlazorWebView/src/SharedSource/BlazorWebViewInitializingEventArgs.cs

```diff
@@ -0,0 +1,68 @@
+﻿using System;
+using System.Threading.Tasks;
+#if WEBVIEW2_WINFORMS
+using Microsoft.Web.WebView2.Core;
+using WebView2Control = Microsoft.Web.WebView2.WinForms.WebView2;
+#elif WEBVIEW2_WPF
+using Microsoft.Web.WebView2.Core;
+using WebView2Control = Microsoft.Web.WebView2.Wpf.WebView2;
+#elif WINDOWS && WEBVIEW2_MAUI
+using Microsoft.Web.WebView2.Core;
+using WebView2Control = Microsoft.UI.Xaml.Controls.WebView2;
+#elif ANDROID
+using AWebView = Android.Webkit.WebView;
+#elif IOS || MACCATALYST
+using WebKit;
+#endif
+
+namespace Microsoft.AspNetCore.Components.WebView
+{
+	/// <summary>
+	/// Allows configuring the underlying web view when the application is initializing.
+	/// </summary>
+	public partial class BlazorWebViewInitializingEventArgs : EventArgs
+	{
+#nullable disable
+#if WINDOWS
+		/// <summary>
+		/// Gets or sets the browser executable folder path for the <see cref="WebView2Control"/>.
+		/// </summary>
+		public string BrowserExecutableFolder { get; set; }
+
+		/// <summary>
+		/// Gets or sets the user data folder path for the <see cref="WebView2Control"/>.
+		/// </summary>
+		public string UserDataFolder { get; set; }
+
+		/// <summary>
+		/// Gets or sets the environment options for the <see cref="WebView2Control"/>.
+		/// </summary>
+		public CoreWebView2EnvironmentOptions EnvironmentOptions { get; set; }
+#endif
+
+#if WINDOWS
+		/// <summary>
+		/// Gets or sets a function that will be invoked once the web view has been initialized with
+		/// the default values to allow further configuring additional options.
+		/// </summary>
+		public Action<WebView2Control> OnWebViewInitialized { get; set; }
```

Comment:
The more I look at this, I feel it's weird to have an event on an event args. Why not have two events, one for before, and one for after? Especially for platforms like Android where there is no "before" at all. You have to hook it up just to set up an after event, and then put your real code in the after event. I would have InitializINGEventArgs and InitializEDEventArgs with matching events. For some platforms the args might be quite empty for now (because there's no args to provide), but in the future of course we might add something. So that way the events are consistent on all platforms, even though the event args are quite different (which is already the case in this PR, which I think is totally fine).

Reply:
That's fair, I don't feel strongly about it, so I think I'll change it for a pair of events.

Reply:
Switch to using 2 separate events